### ISSUE TITLE
Change interactive mode word storage so everything that's typed is saved

### ISSUE THREAD

**Issue by misterzenzel** [cmsc22000-project-2018/autocomplete#63]
Right now, only the most recently typed word is stored. But, in order to delete multiple words or to save the screen to a file, everything has to be stored.

**Comment by misterzenzel** [comment]
Changed the system of how words are constructed as well as how characters in general are stored in order to allow deleting multiple words. Previously, only the last prefix/word was kept in storage, allowing for easy construction of a char* prefix to give to the autocomplete function from a linked list of characters and to not have to worry about unexpected characters being passed in. 

In order to fix this, we had several ideas/implementations. Our first was to have a singly linked list of linked lists of characters (the prefixes). To improve the speed we thought we should use a doubly linked list instead so that we could traverse backwards, rather than forwards everytime the delete button was pressed. This implementation we thought would be slow, and cumbersome to correct to the last autocompleted word. In practice, it was not nice to implement.

Our next and final idea was to have a single long linked list of every character that is pressed. The challenge here was changing how we constructed the linked list of characters into a char* prefix to pass to autocomplete. One idea was to have parallel linked lists of characters and word lengths, but that is terrible programming practice, slow, and unnecessarily complex. Instead, we added an element to the node struct of our linked list that kept track of the length of a word at each prefix.

In the end our implementation looks like the following:
Everytime a key is pressed, add it to the linked list. If that character is not a valid autocomplete input (aka a-z/A-Z), then assign the length variable of the node to -1. Otherwise, until a non valid autocomplete input is pressed, assigning a counting length. This also allowed us to create a clever word construction algorithm in which a char* is constructed from the characters last inputted/not deleted in the linked list of size, length in the first node we pop off. So at any given point in the word, we know how long the prefix we're at is.
